controllers/deployment.go

```diff
@@ -172,8 +175,17 @@ func buildStatefulSets(master *v1alpha1.CDAPMaster, name string, services Servic
 	spec = spec.withInitContainer(
 		newContainerSpec(master, "StorageInit", dataDir).setArgs(containerStorageMain))
 
+	metricsSidecarEnabled := true
```

Comment:
The ServiceGroupInfo should already have the info for whether metrics collection is enabled, only additionally check if individual service spec has overwritten this value.

Reply:
Also, it may be better to make this check after fetching the CDAPServiceSpec inside the for loop.